### DISCUSSION TITLE
How to send a Screenshot to another API/backend after taking the screenshot?

### DISCUSSION THREAD

**Discussion by mihirm3hub** [google/model-viewer#4109] [Q&A]
So I am looking for a screenshot event after the screenshot button is clicked. And then send the captured screenshot image to a backend server how can that be achieved?
I did find the [toDataURL](https://modelviewer.dev/docs/#entrydocs-loading-methods-toDataURL:~:text=toDataURL(type%2C%20encoderOptions)) method but did not find any way to listen to click events for the screenshot button which is native to the mv ar experience.
Thank you for any help.

**Comment by elalish** [comment]
It sounds like you're talking about SceneViewer, which is a separate app that MV can't control, so I don't think that's possible. It saves the screenshots to the photos reel, so perhaps you could make a UI to upload recent photos instead?

**Reply by mihirm3hub** [reply]
@elalish  thanks for the reply. Any way I can create the UI as there seems to be no access to the SceneViewer events so how do I show the user the UI once they click the snapshot button?

**Reply by elalish** [reply]
That's a separate app, so you'll have to wait until they close it and come back to your website. I don't think there's any way to know if they took screenshots, but you can give them the option to upload photos and explain what they should do. In WebXR mode you'd have more options, but since we haven't implemented screen shots yet, you'd need to get the camera image and composite it yourself.
